nilearn/glm/first_level/first_level.py

```diff
@@ -961,105 +951,441 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
+
+    Parameters
+    ----------
+    dataset_path : :obj:`str`
+        Fullpath of the BIDS dataset root folder.
+
+    task_label : :obj:`str`
+        Task_label as specified in the file names like _task-<task_label>_.
+
+    space_label : :obj:`str`
+        Specifies the space label of the preprocessed bold.nii images.
+        As they are specified in the file names like _space-<space_label>_.
+
+    img_filters : :obj:`list` of :obj:`tuples` (str, str)
+        Filters are of the form (field, label).
+        Only one filter per field allowed.
+
+    derivatives_path : :obj:`str`
+        Fullpath of the BIDS dataset derivative folder.
+
+    """
+    if not isinstance(dataset_path, str):
+        raise TypeError(
+            "'dataset_path' must be a string. "
+            f"Got {type(dataset_path)} instead."
+        )
+    if not os.path.exists(dataset_path):
+        raise ValueError(f"'dataset_path' does not exist:\n{dataset_path}")
+
+    if not os.path.exists(derivatives_path):
+        raise ValueError("derivatives folder not found in given dataset:\n"
+                         f"{derivatives_path}")
+
+    if not isinstance(task_label, str):
+        raise TypeError(
+            "'task_label' must be a string. "
+            f"Got {type(task_label)} instead."
+        )
+    if not _is_alphanumeric(task_label):
+        raise ValueError(
+            "'task_label' must be alphanumeric. "
+            f"Got {task_label} instead."
+        )    
+
+    if space_label is not None:
+        if not isinstance(space_label, str):
+            raise TypeError(
+                "'space_label' must be a string, "
+                f"Got {type(space_label)} instead."
+            )
+        if not _is_alphanumeric(space_label):
+            raise ValueError(
+                "'space_label' must be alphanumeric. "
+                f"Got {space_label} instead."
+            )      
+
+    if not isinstance(img_filters, list):
+        raise TypeError(
+            f"'img_filters' must be a list. "
+            f"Got {type(img_filters)} instead."
+        )
+
+    supported_filters = _supported_bids_filter()["raw"] +\
+                        _supported_bids_filter()["derivatives"]
+    for filter_ in img_filters:
+
+        if not isinstance(filter_[0], str):
+            raise TypeError('Filters in img_filters must be (str, str). '
+                            f"Got {type(filter_[0])} for {filter_} instead.")
+
+        if filter_[0] not in supported_filters:
+            raise ValueError(
+                f"Entity {filter_[0]} for {filter_} is not a possible filter. "
+                f"Only {supported_filters} are allowed.")        
+
+        if (not isinstance(filter_[1], str) 
+            or not _is_alphanumeric(filter_[1])):
+            raise TypeError('BIDS labels in img_filters must be alphanumeric. '
+                            f"Got {type(filter_[1])} for {filter_} instead.")
+
+
+def _is_alphanumeric(string: str) -> bool:
```

Comment:
I opine move it to a module in `_utils/` package.
And `_utils` is already protected, not meant for users to call, so the name can be `is_alphanumeric()`

Reply:
inlined the code for now
we can refactor this in another PR